src/compiler/transformers/esDecorators.ts

```diff
@@ -2332,4 +2340,59 @@ export function transformESDecorators(context: TransformationContext): (x: Sourc
             factory.createStringLiteral("default") :
             factory.createStringLiteralFromNode(name);
     }
+
+    function createMetadata(name: Identifier, hasProto: boolean) {
+        const varDecl = factory.createVariableDeclaration(
+            name,
+            /*exclamationToken*/ undefined,
+            /*type*/ undefined,
+            factory.createConditionalExpression(
+                factory.createBinaryExpression(
+                    factory.createBinaryExpression(factory.createTypeOfExpression(factory.createIdentifier("Symbol")), factory.createToken(SyntaxKind.EqualsEqualsEqualsToken), factory.createStringLiteral("function")),
+                    factory.createToken(SyntaxKind.AmpersandAmpersandToken),
+                    factory.createPropertyAccessExpression(factory.createIdentifier("Symbol"), factory.createIdentifier("metadata")),
+                ),
+                factory.createToken(SyntaxKind.QuestionToken),
+                factory.createCallExpression(
+                    factory.createPropertyAccessExpression(factory.createIdentifier("Object"), factory.createIdentifier("create")),
+                    /*typeArguments*/ undefined,
+                    [hasProto ? createSymbolMetadataReference() : factory.createNull()]
+                ),
+                factory.createToken(SyntaxKind.ColonToken),
+                factory.createIdentifier("undefined"),
+            ),
+        );
+        return factory.createVariableStatement(/*modifiers*/ undefined, factory.createVariableDeclarationList([varDecl], NodeFlags.Const));
+    }
+
+    function createSymbolMetadata(target: Identifier | ThisExpression, value: Identifier) {
+        const properties = [
+            factory.createPropertyAssignment("configurable", factory.createTrue()),
+            factory.createPropertyAssignment("writable", factory.createTrue()),
+            factory.createPropertyAssignment("enumerable", factory.createTrue()),
+            factory.createPropertyAssignment("value", value),
+        ];

```

Comment:
You can replace this with a call to `factory.createPropertyDescriptor({ configurable: true, writable: true, enumerable: true, value })`